scripts/autotools/libmongocrypt/CheckSSL.m4

```diff
@@ -0,0 +1,35 @@
+if test "$PHP_CLIENT_SIDE_ENCRYPTION" != "no"; then
+  AC_MSG_CHECKING([which crypto library to use for libmongocrypt])
+  AC_MSG_RESULT([$PHP_MONGODB_SSL])
+
+  dnl Disable Windows crypto
+  AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_CNG, 0)
+
+  if test "$PHP_MONGODB_SSL" = "darwin"; then
+    AC_SUBST(MONGOC_ENABLE_CLIENT_SIDE_ENCRYPTION, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_OPENSSL, 0)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_COMMON_CRYPTO, 1)
+
+    PHP_MONGODB_ADD_SOURCES([src/libmongocrypt/src/], "crypto/commoncrypto.c", $PHP_MONGODB_LIBMONGOCRYPT_CFLAGS)
+    PHP_MONGODB_ADD_SOURCES([src/libmongocrypt/kms-message/src/], "kms_crypto_apple.c", $PHP_MONGODB_LIBMONGOCRYPT_CFLAGS)
+  elif test "$PHP_MONGODB_SSL" = "openssl" -o "$PHP_MONGODB_SSL" = "libressl"; then
+    AC_SUBST(MONGOC_ENABLE_CLIENT_SIDE_ENCRYPTION, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_OPENSSL, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_COMMON_CRYPTO, 0)
+
+    PHP_MONGODB_ADD_SOURCES([src/libmongocrypt/src/], "crypto/openssl.c", $PHP_MONGODB_LIBMONGOCRYPT_CFLAGS)
+    PHP_MONGODB_ADD_SOURCES([src/libmongocrypt/kms-message/src/], "kms_crypto_openssl.c", $PHP_MONGODB_LIBMONGOCRYPT_CFLAGS)
+  elif test "$PHP_CLIENT_SIDE_ENCRYPTION" = "auto"; then
```

Comment:
Can you remind me if we decided to prohibit compilation of libmongocrypt altogether if a crypto library wasn't available, rather than allow it to build and raise a runtime error?

Reply:
See https://github.com/mongodb/mongo-php-driver/pull/1065#discussion_r356944874:
> If not building with TLS, can we skip defining `MONGOC_ENABLE_CLIENT_SIDE_ENCRYPTION` to
disable FLE in libmongoc instead? That will give a better error to the user, along the lines of "Client Side Encryption not supported" if they attempt to use FLE.

When compiling in auto mode, we can silently disable FLE, but when explicitly compiling with libmongocrypt we will error at configure-time as the user explicitly wants to build with FLE. We can remove the error message and rely on errors when running code, but I thought it was more prudent to cancel at compile time to make it explicit.

Reply:
Compile time error works for me, especially since the default `pecl install` case will just use `auto`.